src/main/java/org/springframework/data/elasticsearch/core/ReactiveDocumentOperations.java

```diff
@@ -78,6 +86,74 @@
 	 */
 	<T> Mono<T> save(T entity, IndexCoordinates index);
 
+	/**
+	 * Index entities under the given {@literal type} in the given {@literal index}. If the {@literal index} is
+	 * {@literal null} or empty the index name provided via entity metadata is used. Same for the {@literal type}.
+	 *
+	 * @param entities must not be {@literal null}.
+	 * @param index the target index, must not be {@literal null}
+	 * @param <T>
+	 * @return a {@link Flux} emitting saved entities.
+	 */
+	default <T> Flux<T> saveAll(Iterable<T> entities, IndexCoordinates index) {
+		return saveAll(Flux.fromIterable(entities), index);
+	}
+
+	/**
+	 * Index entities under the given {@literal type} in the given {@literal index}. If the {@literal index} is
+	 * {@literal null} or empty the index name provided via entity metadata is used. Same for the {@literal type}.
+	 *
+	 * @param entities must not be {@literal null}.
+	 * @param index the target index, must not be {@literal null}
+	 * @param <T>
+	 * @return a {@link Flux} emitting saved entities.
+	 */
+	<T> Flux<T> saveAll(Publisher<T> entities, IndexCoordinates index);
+
+	/**
+	 * Execute a multiGet against elasticsearch for the given ids.
+	 *
+	 * @param query the query defining the ids of the objects to get
+	 * @param clazz the type of the object to be returned
+	 * @param index the index(es) from which the objects are read.
+	 * @return flux with list of nullable objects
+	 */
+	<T> Flux<T> multiGet(Query query, Class<T> clazz, IndexCoordinates index);
+
+	/**
+	 * Bulk index all objects. Will do save or update.
+	 *
+	 * @param queries the queries to execute in bulk
+	 */
+	default Mono<Void> bulkIndex(List<IndexQuery> queries, IndexCoordinates index) {
```

Comment:
We should not expose methods accepting `IndexQuery` yet. The `IndexQuery` is one of the types that requires further refinement until it can be useful (e.g. type information are lost `IndexQuery` vs `IndexQuer<T>`). We also should consider whether we require `IndexQuery` at all or whether we can replace `IndexQuery` with the more generic `Document` API approach (this isn't related to this PR, but belongs still into the context when talking about `IndexQuery`).